### ISSUE TITLE
Properly notify the user when premium sync was not successful for extended periods of time

### ISSUE THREAD

**Issue by kelsos** [rotki/rotki#7375]
## Description

The message on database failures as we discussed is designed to be temporary in message and only indicates that there was a temporary in nature upload failure. This means that the situation is not critical and it might recover in a future attempt.

We need a new way to indicate the critical situation where periodic database sync has been failing for an extended period of time. The scenario we would like to avoid is the case where a user believes that the latest database is synced only to realize that the latest backup is from days ago.

For this we need a different indication that would be critical and it would show that no backup has been successful for an extended period of time.

This would show in the sync indicator with a red color indicating a critical condition that required the user's attention.

**Comment by kelsos** [comment]
## Implementation plan

The frontend cannot answer this on its own. A stale `last_data_upload_ts` is ambiguous: it looks identical whether uploads are failing or whether there was simply nothing to upload. Resolving that needs exactly one new piece of backend state.

### Backend: one new persisted field

`DBCacheStatic.LAST_DATA_UPLOAD_FAILURE_TS`, the timestamp at which the current genuine failure streak began. `0`/absent means healthy. It is set on the first genuine failure, left untouched on subsequent ones, and cleared on success and on benign no-ops. It lives in `key_value_cache`, so no DB migration is needed and the value survives restarts.

A streak start timestamp is preferred over a failure counter because the condition we care about is duration ("the latest backup is from days ago"), and a counter cannot express duration when the app spends most of its time closed.

#### Classifying the exit paths

`maybe_upload_data_to_server` (`rotkehlchen/premium/sync.py`) has eight exits. The existing `actionable` flag is not a usable classifier here, since `Remote database is up to date` is currently `actionable: True`.

| Exit | Streak |
|---|---|
| Upload succeeded | clear |
| `Remote database is up to date` (same hash) | clear |
| `Remote database is more recent than local` | clear |
| Local database failed the integrity check | start/keep |
| Exported database failed the integrity check | start/keep |
| Fetching metadata error | start/keep |
| `upload_data` RemoteError / PremiumAuthenticationError | start/keep |
| Size discrepancy, user asked | start/keep |

The two benign no-ops must clear the streak. In both cases the remote copy is current, so a stale timestamp is correct and must not raise an alarm. This is the specific false positive that makes a frontend only implementation wrong.

#### The silent outage path

`check_if_should_sync` calls `premium.get_capabilities()`, which performs a network request via `fetch_limits()`. On `RemoteError` or `PremiumAuthenticationError` it logs and returns `False`, so `maybe_upload_data_to_server` never runs and **no `database_upload_result` message is emitted at all**. A premium server outage or an expired API key therefore produces complete silence, which is precisely the scenario this issue exists to catch.

That branch must both start the streak and emit a `database_upload_result` message (`uploaded: false`, `actionable: false`, with a message naming the unreachable server). This is a deliberate behaviour change: silence becomes a signal. It also fixes the existing transient warning, which shows nothing during an outage today.

`max_backup_size_mb == 0` remains a non failure. That means cloud backup is not part of the user's plan, and the frontend already knows it through `PremiumFeature.CLOUD_BACKUP`.

#### Files

- `rotkehlchen/db/cache.py`: new `DBCacheStatic` member.
- `rotkehlchen/premium/sync.py`: `_mark_upload_failure()` / `_clear_upload_failure()` writing the cache key plus an in memory mirror, initialised from the DB in `__init__` beside `last_data_upload_ts`; called per the table above and in the `check_if_should_sync` except branch, together with the new message emit.
- `rotkehlchen/rotkehlchen.py`: expose the field in `query_periodic_data`.
- `docs/websockets.rst` and the API docs: the new `/periodic` field and the new emit point.

The existing `/periodic` `last_data_upload_ts` is left untouched. It carries `last_remote_data_upload_ts`, which is the correct value for the "Last data upload" display since another device may have written the backup.

### Frontend

- Add the field to `PeriodicClientQueryResultSchema` and carry it through `use-session-metadata-store` and `use-periodic-data-fetcher`, in the same shape as `lastDataUpload`.
- Track whether an upload attempt has completed in the current session. A plain in memory ref in `use-session-sync`, set by `createDbUploadResultHandler` on any result, success or failure. It must not be `useSessionStorage`, since it has to reset on reload. A manual force sync counts as an attempt.
- New `use-sync-health` composable deriving `ok` / `warning` / `critical`. Keeping it out of the component makes the threshold logic testable with fake timers.
- `SyncIndicator.vue`: the `RuiBadge` colour and the `lu-cloud-off-fill` icon become `error` in the critical state; the tooltip and menu body gain a critical variant naming the date of the last good backup. `SyncUploadStatusAlert.vue` gains an `error` type.
- The critical state derives from the backend field, never from `uploadStatus`. That one stays in `sessionStorage` with its `uploadStatusAlreadyHandled` latch governing only the transient warning. Otherwise a single dismissal would hide the critical state as well.

### Behaviour

Critical is `attemptSeenThisSession && failureSince > 0 && now - failureSince > 24h`.

- **Threshold is 24h** of continuous failure. Automatic attempts are throttled to once per hour, so this is roughly 24 consecutive failed attempts when the app stays open.
- **It does not trigger at login.** The red state appears only after the first upload attempt of the session has completed. Since `last_upload_attempt_ts` is initialised from the last successful upload timestamp, after a login that follows a stale period the throttle has already expired and the attempt fires on the next task manager tick.
- **A single isolated failure never goes red.** If the streak began with this session's own attempt, its age is close to zero and the 24h test fails on its own. A red badge therefore implies at least one earlier failure by construction, with no extra state required.
- **No notification and no task centre entry.** The red indicator alone is the surface.
- A user whose local database genuinely never changes sits at `Remote database is up to date`, which clears the streak. That case stays silent, correctly: their remote copy matches.

### Tests

- Backend: streak transitions across all eight exits, the capabilities error path, and persistence across a `PremiumSyncManager` re-instantiation.
- Frontend: `use-sync-health` covering the session attempt gate, the single failure case, and the 24h boundary.
